tests/integration/artifact_exclusions_integration_test.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+set -eux
+set -o pipefail
+
+readonly SCRIPT_DIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+
+# try to run it once here to bootstrap
+$SCRIPT_DIR/../../third_party/coursier/coursier
+
+# Get list of inputs to the artifact library, which is a transitive exports of deps
+bazel query @other_maven//:com_google_guava_guava_lib --output=xml | grep "rule-input" | cut -d"\"" -f2 | cut -d":" -f2 > /tmp/without_exclusions.txt
+
+# Get list of inputs to the artifact library with excluded artifacts
+bazel query @other_maven_with_exclusions//:com_google_guava_guava_lib --output=xml | grep "rule-input" | cut -d"\"" -f2 | cut -d":" -f2 > /tmp/with_exclusions.txt
+
+# Get the diff between the two
+diff /tmp/with_exclusions.txt /tmp/without_exclusions.txt > /tmp/exclusion_diff.txt || true
+
+# Assert that the diff matches the two excluded artifacts
+diff /tmp/exclusion_diff.txt $SCRIPT_DIR/artifact_exclusions.golden
```

Comment:
Will the contents of /tmp/with_exclusions.txt have the excluded files in the same order each time? Otherwise the test will be flaky

Reply:
good point, sorted the results before writing it.